v2/astradb-to-bigquery/src/main/java/com/google/cloud/teleport/v2/astradb/templates/AstraDbToBigQuery.java

```diff
@@ -0,0 +1,254 @@
+    description = "A batch pipeline which reads a table from AstraDB and copy it to BigQuery.",
+    optionsClass = AstraDbToBigQuery.Options.class,
+    flexContainerName = "astradb-to-bigquery",
+    documentation =
+        "https://cloud.google.com/dataflow/docs/guides/templates/provided/astradb-to-bigquery",
+    contactInformation = "https://cloud.google.com/support")
+public class AstraDbToBigQuery {
+
+  /** Logger for the class. */
+  private static final Logger LOGGER = LoggerFactory.getLogger(AstraDbToBigQuery.class);
+
+  /** If not provided, it is the default token range value. */
+  public static final int DEFAULT_TOKEN_RANGE = 18;
+
+  /**
+   * Options for the sample
+   *
+   * <p>Inherits standard configuration options.
+   */
+  public interface Options
+      extends PipelineOptions,
+          AstraDbToBigQueryOptions.AstraDbSourceOptions,
+          AstraDbToBigQueryOptions.BigQueryWriteOptions {}
+
+  /** Main operations. */
+  public static void main(String[] args) {
+    UncaughtExceptionLogger.register();
+    LOGGER.info("Starting pipeline");
+
+    try {
+
+      Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
+      LOGGER.debug("Pipeline Arguments (options) validated");
+
+      // --------------------------------
+      // AstraDbIO.Read<Row>
+      // --------------------------------
+
+      // Credentials are read from secrets manager
+      AbstractMap.SimpleImmutableEntry<String, byte[]> astraCredentials =
+          parseAstraCredentials(options);
+      LOGGER.debug("Astra Credentials parsed");
+
+      // Map Cassandra Table Schema into BigQuery Table Schema
+      SerializableFunction<AstraDbIO.Read<?>, TableSchema> bigQuerySchemaFactory =
+          new AstraDbToBigQueryMappingFn(options.getAstraKeyspace(), options.getAstraTable());
+      LOGGER.debug("Schema Mapper has been initialized");
+
+      // Map Cassandra Rows into (Apache) Beam Rows (DATA)
+      SerializableFunction<CqlSession, AstraDbMapper<Row>> beamRowMapperFactory =
+          new BeamRowDbMapperFactoryFn(options.getAstraKeyspace(), options.getAstraTable());
+      LOGGER.debug("Row Mapper has been initialized");
+
+      // Distribute reads across all available Cassandra nodes
+      int minimalTokenRangesCount =
+          (options.getMinTokenRangesCount() == null)
+              ? DEFAULT_TOKEN_RANGE
+              : options.getMinTokenRangesCount();
+
+      // Source: AstraDb
+      AstraDbIO.Read<Row> astraSource =
+          AstraDbIO.<Row>read()
+              .withToken(astraCredentials.getKey())
+              .withSecureConnectBundle(astraCredentials.getValue())
+              .withKeyspace(options.getAstraKeyspace())
+              .withTable(options.getAstraTable())
+              .withMinNumberOfSplits(minimalTokenRangesCount)
+              .withMapperFactoryFn(beamRowMapperFactory)
+              .withCoder(SerializableCoder.of(Row.class))
```

Comment:
Can we use RowCoder? It is much better / faster compared to SerializableCoder

Reply:
The would required more investigation and the change of `AstraDb.Read<>` i will investigate but better if the PR can be merge with the `SerializableCoder`